engine/src/main/java/org/entando/entando/aps/system/services/digitalexchange/marketplace/MarketplacesServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.digitalexchange.marketplace;
+
+import com.agiletec.aps.system.SystemConstants;
+import com.agiletec.aps.system.exception.ApsSystemException;
+import com.agiletec.aps.system.services.baseconfig.ConfigInterface;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.xml.bind.DataBindingException;
+import javax.xml.bind.JAXB;
+import org.entando.entando.aps.system.exception.RestRourceNotFoundException;
+import org.entando.entando.aps.system.services.digitalexchange.marketplace.model.Marketplace;
+import org.entando.entando.aps.system.services.digitalexchange.marketplace.model.MarketplacesConfig;
+import org.entando.entando.web.common.exceptions.ValidationConflictException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.validation.BeanPropertyBindingResult;
+import static org.entando.entando.web.digitalexchange.marketplace.MarketplaceValidator.*;
+
+public class MarketplacesServiceImpl implements MarketplacesService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+
+    private ConfigInterface configManager;
+
+    public ConfigInterface getConfigManager() {
+        return configManager;
+    }
+
+    public void setConfigManager(ConfigInterface configManager) {
+        this.configManager = configManager;
+    }
+
+    @Override
+    public List<Marketplace> getMarketplaces() {
+
+        String marketplacesConfig = configManager.getConfigItem(SystemConstants.CONFIG_ITEM_MARKETPLACES);
+
+        if (marketplacesConfig == null) {
+            logger.warn("Marketplaces configuration not found");
+            return new ArrayList<>();
+        }
+
+        try {
+            MarketplacesConfig config = JAXB.unmarshal(new StringReader(marketplacesConfig), MarketplacesConfig.class);
+            return config.getMarketplaces();
+        } catch (DataBindingException ex) {
+            logger.error("Unable to parse Marketplaces XML configuration", ex);
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public Marketplace findByName(String name) {
+        List<Marketplace> marketplaces = getMarketplaces();
+        Optional<Marketplace> maybeMarketplace = getMarketplace(marketplaces, name);
+        if (!maybeMarketplace.isPresent()) {
+            throw new RestRourceNotFoundException(ERRCODE_MARKETPLACE_NOT_FOUND, "marketplace", name);
+        }
+        return maybeMarketplace.get();
```

Comment:
Your code here is correct but if you prefer to avoid the isPresent test:

`        return maybeMarketplace.orElseThrow(
                () -> new RestRourceNotFoundException(
                        ERRCODE_MARKETPLACE_NOT_FOUND, 
                        "marketplace", 
                        name
                )
        );
`